test/metabase/sync_database/analyze_test.clj

```diff
@@ -72,3 +79,77 @@
+
+(defn get-latest-sync-time [db_id table-name]
+  (let [table-id (db/select-one-id Table, :db_id db_id, :active true :name table-name)]
+    (db/select-field :last_analyzed Field :table_id table-id)))
+
+(defn set-table-visibility-type! [table state]
+  ((user->client :crowberto) :put 200 (format "table/%d" (:id table)) {:display_name    "hiddentable"
+                                                                       :entity_type     "person"
+                                                                       :visibility_type state
+
+                                                                       :description     "What a nice table!"}))
+(defn api-sync-call [table]
+  ((user->client :crowberto) :post 200 (format "database/%d/sync" (:db_id table)) {}))
+
+(defn sync-call [table]
+  (let [db-id (:db_id table)]
+    (analyze-data-shape-for-tables! (driver/database-id->driver db-id) {:id db-id})))
+
+;; expect all the kinds of hidden tables to stay un-analyzed through transitions and repeated syncing
+(tt/expect-with-temp [Table [table {:rows 15}]
+                      Field [field {:table_id (:id table)}]]
+  1
+  (do (set-table-visibility-type! table "hidden")
+      (api-sync-call table)
+      (set-table-visibility-type! table "cruft")
+      (set-table-visibility-type! table "cruft")
+      (api-sync-call table)
+      (set-table-visibility-type! table "technical")
+      (api-sync-call table)
+      (set-table-visibility-type! table "technical")
+      (api-sync-call table)
+      (api-sync-call table)
+      (count-unanalyzed-fields (:db_id table) (:name table))))
+
+;; same test not coming through the api
+(tt/expect-with-temp [Table [table {:rows 15}]
+                      Field [field {:table_id (:id table)}]]
+  1
+  (do (set-table-visibility-type! table "hidden")
+      (sync-call table)
+      (set-table-visibility-type! table "cruft")
+      (set-table-visibility-type! table "cruft")
+      (sync-call table)
+      (set-table-visibility-type! table "technical")
+      (sync-call table)
+      (set-table-visibility-type! table "technical")
+      (sync-call table)
+      (sync-call table)
+      (count-unanalyzed-fields (:db_id table) (:name table))))
+
+;; un-hiding a table should cause it to be analyzed
+(tt/expect-with-temp [Table [table {:rows 15}]
+                      Field [field {:table_id (:id table)}]]
+  0
+  (do (set-table-visibility-type! table "hidden")
+      (set-table-visibility-type! table nil)
+      (count-unanalyzed-fields (:db_id table) (:name table))))
+
+;; re-hiding a table should not cause it to be analyzed
+(tt/expect-with-temp [Table [table {:rows 15}]
+                      Field [field {:table_id (:id table)}]]
+  (do (set-table-visibility-type! table "hidden")
+      (set-table-visibility-type! table nil)
+      (get-latest-sync-time (:db_id table) (:name table)))
+  (do (set-table-visibility-type! table "hidden")
+      (get-latest-sync-time (:db_id table) (:name table))))
```

Comment:
`(expect actual)` with one form is the same as `(expect true actual)`

and since the temp objects aren't needed for the `expected` form we should push `with-temp` down into `actual`